gson/src/main/java/com/google/gson/JsonValue.java

```diff
@@ -0,0 +1,51 @@
+package com.google.gson;
+
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * A class representing a raw JSON value. This should be used with
+ * {@link JsonWriter#jsonValue(String)} and should not be encoded or escaped further.
+ *
+ * @author Zac Sweers
+ * @since 2.8.7
+ */
+public final class JsonValue extends JsonElement {
+
+  private final String jsonValue;
+
+  public JsonValue(String jsonValue) {
+    this.jsonValue = jsonValue;
```

Comment:
The project is not very consistent with it, but maybe it would be good to check for `null` and throw a `NullPointerException` then?